stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -419,6 +501,58 @@ def apply_datetime_filter(search: Search, datetime_search):
                 "range", properties__datetime={"gte": datetime_search["gte"]}
             )
         return search
+    
+    @staticmethod
+    def apply_datetime_collections_filter(search: Search, datetime_search):
+        """Apply a filter to search collections based on datetime field.
+
+        Args:
+            search (Search): The search object to filter.
+            datetime_search (dict): The datetime filter criteria.
+
+        Returns:
+            Search: The filtered search object.
+        """
+        if "eq" in datetime_search:
+            search = search.filter("range", **{"collection_start_time": {"lte": datetime_search['eq']}})
+            search = search.filter("range", **{"collection_end_time": {"gte": datetime_search['eq']}})            
+
+        else:
+            should = []
+            should.extend(
+                [
+                    Q("bool", filter=[
```

Comment:
Formatting looks weird here, have you run black over this?

Reply:
I hadn't but have now. I think it looks a bit better now.